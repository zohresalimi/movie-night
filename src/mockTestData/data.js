import React, { useReducer } from "react";
import AppContext from "../store/context";
import reducer from "../store/reducer";

export const getTestState = () => {
  return {
    activePage: 1,
    movies: {
      1: {
        page: 1,
        total_pages: 2,
        total_results: 2,
        results: [
          {
            id: 10523,
            adult: false,
            poster_path: "/nJm8UF4zbUYEoku9seEtz1PTJGS.jpg",
          },
          {
            id: 50647,
            adult: false,
            poster_path: "/PlzkOA19mxEit7tV3AUnBYZ0io.jpg",
          },
        ],
      },
    },
    watchLaterList: {
      10523: {
        id: 10523,
        adult: false,
        poster_path: "/nJm8UF4zbUYEoku9seEtz1PTJGS.jpg",
      },
    },
    favoriteList: {
      10523: {
        id: 10523,
        adult: false,
        poster_path: "/nJm8UF4zbUYEoku9seEtz1PTJGS.jpg",
      },
    },
    apiConfig: {
      secure_base_url: "https://image.tmdb.org/t/p/",
      poster_sizes: ["w92", "w154", "w185", "w342", "w500", "w780"],
    },
  };
};

export const WithProvider = (props) => {
  const [state, dispatch] = useReducer(
    reducer,
    props.defaultValue || getTestState()
  );

  return (
    <AppContext.Provider
      value={{
        state,
        dispatch: props.mockDispatch || dispatch,
      }}
    >
      {props.children}
    </AppContext.Provider>
  );
};
