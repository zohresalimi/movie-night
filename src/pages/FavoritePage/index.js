import React, { useContext } from "react";
import { Grid, Segment, Header, Icon, Button } from "semantic-ui-react";

import AppContext from "../../store/context";
import CardItem from "../../components/CardItem";

export default function FavoritePage() {
  const { state } = useContext(AppContext);
  const { favoriteList } = state;

  return (
    <Grid>
      <Grid.Row data-testid="favorite-page-wrapper">
        {!Object.keys(favoriteList).length ? (
          <Grid.Column>
            <Segment placeholder>
              <Header icon>
                <Icon name="film" />
                Favorite list is empty
              </Header>
            </Segment>
          </Grid.Column>
        ) : (
          Object.keys(favoriteList).map((item) => (
            <Grid.Column mobile={16} tablet={8} computer={4} key={item}>
              <CardItem item={favoriteList[item]} />
            </Grid.Column>
          ))
        )}
      </Grid.Row>
    </Grid>
  );
}
