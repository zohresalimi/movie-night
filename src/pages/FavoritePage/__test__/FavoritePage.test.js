import * as React from "react";
import { act, render } from "@testing-library/react";

import FavoritePage from "..";
import { WithProvider, getTestState } from "../../../mockTestData/data";

async function renderWrapper(props) {
  let component;

  act(() => {
    component = render(
      <WithProvider {...props}>
        <FavoritePage />
      </WithProvider>
    );
  });

  return component;
}

describe("Favorite Page Component Testing", () => {
  test("loads items", async () => {
    const { container, getByTestId } = await renderWrapper();
    expect(getByTestId("favorite-page-wrapper")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test("shows message with icon if favorite list is empty", async () => {
    const testState = getTestState();
    const { container } = await renderWrapper({
      defaultValue: {
        ...testState,
        favoriteList: {},
      },
    });

    expect(container).toMatchSnapshot();
  });
});
