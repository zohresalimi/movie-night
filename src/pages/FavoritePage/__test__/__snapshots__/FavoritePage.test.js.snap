// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Page Component Testing loads items 1`] = `
<div>
  <div
    class="ui grid"
  >
    <div
      class="row"
      data-testid="favorite-page-wrapper"
    >
      <div
        class="four wide computer sixteen wide mobile eight wide tablet column"
      >
        <div
          class="card-wrapper"
        >
          <div
            class="ui card"
          >
            <div
              class="small image dimmable"
            >
              <div
                class="ui dimmer"
              >
                <div
                  class="content"
                >
                  <div>
                    <button
                      data-testid="play-btn"
                    >
                      <a
                        class="sc-bdfBwQ gaGgLj"
                      >
                        <i
                          aria-hidden="true"
                          class="play mini icon"
                        />
                      </a>
                    </button>
                    <div
                      class="tools-wrapper"
                    >
                      <div
                        class="ui small basic vertical buttons"
                      >
                        <button
                          class="ui yellow basic icon button"
                        >
                          <i
                            aria-hidden="true"
                            class="star icon"
                          />
                        </button>
                        <button
                          class="ui yellow basic icon button"
                        >
                          <i
                            aria-hidden="true"
                            class="clock icon"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <img
                src="https://image.tmdb.org/t/p/w500//nJm8UF4zbUYEoku9seEtz1PTJGS.jpg"
              />
            </div>
            <div
              class="content"
            >
              <div
                class="header"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
